x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,479 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { run } from '@kbn/dev-cli-runner';
+import yargs from 'yargs';
+import _ from 'lodash';
+import globby from 'globby';
+import pMap from 'p-map';
+import { ToolingLog } from '@kbn/tooling-log';
+import { withProcRunner } from '@kbn/dev-proc-runner';
+import cypress from 'cypress';
+import grep from '@cypress/grep/src/plugin';
+import crypto from 'crypto';
+import fs from 'fs';
+import { createFailError } from '@kbn/dev-cli-errors';
+import axios from 'axios';
+import path from 'path';
+import os from 'os';
+import pRetry from 'p-retry';
+
+import { renderSummaryTable } from './print_run';
+import type { SecuritySolutionDescribeBlockFtrConfig } from './utils';
+import { parseTestFileConfig, retrieveIntegrations } from './utils';
+
+interface ProductType {
+  product_line: string;
+  product_tier: string;
+}
+
+interface CreateEnvironmentRequestBody {
+  name: string;
+  region_id: string;
+  product_types: ProductType[];
+}
+
+interface Environment {
+  name: string;
+  id: string;
+  region: string;
+  es_url: string;
+  kb_url: string;
+  product: string;
+}
+
+interface Credentials {
+  username: string;
+  password: string;
+}
+
+const DEFAULT_REGION = 'aws-eu-west-1';
+const PROJECT_NAME_PREFIX = 'kibana-cypress-security-solution-ephemeral';
+const BASE_ENV_URL = 'https://global.qa.cld.elstc.co';
+let log: ToolingLog;
+
+const getApiKeyFromElasticCloudJsonFile = (): string | undefined => {
+  const userHomeDir = os.homedir();
+  try {
+    const jsonString = fs.readFileSync(path.join(userHomeDir, '.elastic/cloud.json'), 'utf-8');
+    const jsonData = JSON.parse(jsonString);
+    return jsonData.api_key.qa;
+  } catch (e) {
+    log.info('API KEY could not be found in .elastic/cloud.json');
+  }
+};
+
+// Method to invoke the create environment API for serverless.
+async function createEnvironment(
+  projectName: string,
+  runnerId: string,
+  apiKey: string,
+  ftrConfig: SecuritySolutionDescribeBlockFtrConfig,
+  onEarlyExit: (msg: string) => void
+): Promise<Environment> {
+  log.info(`${runnerId}: Creating environment ${projectName}...`);
+  const environment = {} as Environment;
+  const body = {
+    name: projectName,
+    region_id: DEFAULT_REGION,
+  } as CreateEnvironmentRequestBody;
+
+  const productTypes: ProductType[] = [];
+  ftrConfig?.productTypes?.forEach((t) => {
+    productTypes.push(t as ProductType);
+  });
+  if (productTypes.length > 0) body.product_types = productTypes;
+
+  await axios
+    .post(`${BASE_ENV_URL}/api/v1/serverless/projects/security`, body, {
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+        'Content-Type': 'application/json',
+      },
+    })
+    .then((response) => {
```

Comment:
I suggest to replace `then()` and `catch()` with `try`/`catch` as it's a widespread approach.